packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -2,27 +2,351 @@
+          ).toString("base64")
+        : undefined;
+    appDefinition.appId = teamsAppId;
+
+    try {
+      await AppStudioClient.updateApp(
+        teamsAppId!,
+        appDefinition,
+        appStudioToken,
+        logProvider,
+        colorIconContent,
+        outlineIconContent
+      );
+      return ok(teamsAppId!);
+    } catch (e) {
+      if (e instanceof Error) {
+        return err(
+          returnSystemError(
+            new Error(`Failed to update ${type} teams app manifest due to ${e.name}: ${e.message}`),
+            "Solution",
+            type === "remote"
+              ? SolutionError.FailedToUpdateAppIdInAppStudio
+              : SolutionError.FailedToUpdateLocalAppIdInAppStudio
+          )
+        );
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * ask app common questions to generate app manifest
+   */
+  public async createManifest(settings: ProjectSettings): Promise<TeamsAppManifest | undefined> {
+    const solutionSettings: AzureSolutionSettings =
+      settings.solutionSettings as AzureSolutionSettings;
+    if (
+      !solutionSettings.capabilities ||
+      (!solutionSettings.capabilities.includes(BotOptionItem.id) &&
+        !solutionSettings.capabilities.includes(MessageExtensionItem.id) &&
+        !solutionSettings.capabilities.includes(TabOptionItem.id))
+    ) {
+      throw new Error(`Invalid capability: ${solutionSettings.capabilities}`);
+    }
+    if (
+      HostTypeOptionAzure.id === solutionSettings.hostType ||
+      solutionSettings.capabilities.includes(BotOptionItem.id) ||
+      solutionSettings.capabilities.includes(MessageExtensionItem.id)
+    ) {
+      let manifestString = TEAMS_APP_MANIFEST_TEMPLATE;
+      manifestString = this.replaceConfigValue(manifestString, "appName", settings.appName);
+      manifestString = this.replaceConfigValue(manifestString, "version", "1.0.0");
+      const manifest: TeamsAppManifest = JSON.parse(manifestString);
+      if (solutionSettings.capabilities.includes(TabOptionItem.id)) {
+        manifest.staticTabs = STATIC_TABS_TPL;
+        manifest.configurableTabs = CONFIGURABLE_TABS_TPL;
+      }
+      if (solutionSettings.capabilities.includes(BotOptionItem.id)) {
+        manifest.bots = BOTS_TPL;
+      }
+      if (solutionSettings.capabilities.includes(MessageExtensionItem.id)) {
+        manifest.composeExtensions = COMPOSE_EXTENSIONS_TPL;
+      }
+      return manifest;
+    }
+
+    return undefined;
+  }
+
+  public async reloadManifestAndCheckRequiredFields(
+    ctxRoot: string
+  ): Promise<Result<TeamsAppManifest, FxError>> {
+    const result = await this.reloadManifest(ctxRoot);
+    return result.andThen((manifest) => {
+      if (
+        manifest === undefined ||
+        manifest.name.short === undefined ||
+        manifest.name.short.length === 0
+      ) {
+        return err(
+          returnSystemError(
+            new Error("Name is missing"),
+            "Solution",
```

Comment:
Error names should be changed to some short hand for appstudio plugin instead of solution

Reply:
Already updated the error names.